### DISCUSSION TITLE
todo-Push-Firebase | Default FirebaseApp is not initialized

### DISCUSSION THREAD

**Discussion by quintonv** [shinyorg/shiny#932] [Q&A] [answered]
Hi

I followed the instructions on the sample. I copied google-services.json and created appsettings.json in the same folder as the .sln file for both of these files. When I start the application there's no token in this.pushManager.CurrentRegistrationToken and the test button returns this exception below.

"Default FirebaseApp is not initialized in this process com.shiny.test. Make sure to call FirebaseApp.initializeApp(Context) first."

I tried manually creating the oncreate code by copying from the source generator and then putting FirebaseApp.initializeApp in it, but this didn't help.

Thanks
Quinton

**Comment by aritchie** [comment, marked as answer]
The samples aren't complete for push and use a tool called mobile build tools.  If you are using the sample directly, you still have to call RequestAccess before the current token will be set

**Reply by quintonv** [reply]
Ok thanks, as far as I understood the Test button in the sample app does the Request Access, that is the method that returns the FirebaseApp error. Doesn't this below do what you are saying I should try?

// code from the test button
            this.RequestAccess = this.LoadingCommand(async () =>
            {
                var result = await this.pushManager.RequestAccess();
                this.AccessStatus = result.Status;
                this.Refresh();
            });

**Reply by aritchie** [reply]
Yes but I haven't finished those samples hence the TODO in front of them.  They are currently stubbed and did require additional updates.  You also need to do more than copy an appsettings.json & the google json.

**Reply by quintonv** [reply]
Thanks for the assistance. I know they're not working but thought it might be easy to get it up and running. With the documentation it's tricky to figure it out. Thanks for replying though, I will try figure it again at some stage when I get time, really want to make use of Shiny.

**Reply by aritchie** [reply]
There isn't any difference in setup from a normal app outside of the code.  You still have to do your info.plist & google json regardless of shiny.

**Reply by quintonv** [reply]
Alright, thanks for all the info. Really do appreciate it.
